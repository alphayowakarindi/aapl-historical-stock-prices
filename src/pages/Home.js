import { useSelector } from 'react-redux';
import React from 'react';
import Stock from '../components/Stock';
import Header from '../components/Header';
import Stats from '../components/Stats';

function Home() {
  const range = useSelector((state) => state.range.change);

  let stockList = useSelector((store) => store.stock);

  if (range === 'all') {
    stockList = [...stockList];
  } else if (range === '<1') {
    stockList = [...stockList.filter((stock) => stock.change < 1)];
  } else if (range === '1+') {
    stockList = stockList.filter(
      (stock) => stock.change > 1 && stock.change < 2,
    );
  } else if (range === '2+') {
    stockList = stockList.filter(
      (stock) => stock.change > 2 && stock.change < 3,
    );
  } else {
    stockList = stockList.filter((stock) => stock.change > 3);
  }

  if (stockList.length > 0) {
    return (
      <div className="stocks">
        <Header />
        <Stats />
        <div className="stock-container">
          {stockList.map((stock) => (
            <Stock key={stock.date} date={stock.date} change={stock.change} />
          ))}
        </div>
      </div>
    );
  }
  return (
    <div className="home-loading">
      <span>Loading...</span>
    </div>
  );
}

export default Home;
