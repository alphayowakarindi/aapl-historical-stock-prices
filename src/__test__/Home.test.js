import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import { BrowserRouter } from 'react-router-dom';
import Home from '../pages/Home';

test('renders when the home component matches snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
