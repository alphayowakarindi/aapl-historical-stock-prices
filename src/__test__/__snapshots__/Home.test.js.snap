// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when the home component matches snapshot 1`] = `
<div
  className="stock-container"
/>
`;
